x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/support/ApiKeyUserRoleDescriptorResolver.java

```diff
@@ -37,15 +36,10 @@ public void resolveUserRoleDescriptors(final Authentication authentication, fina
             return;
         }
 
-        rolesStore.getRoleDescriptorsList(effectiveSubject, listener.delegateFailureAndWrap(this::handleRoleDescriptorsList));
+        rolesStore.getRoleDescriptors(effectiveSubject, listener.delegateFailureAndWrap(this::handleRoleDescriptors));
```

Comment:
The `getRoleDescriptorsList` method indeed only returns a single member list today because we do not have proper support for derived API keys (API Keys created by another API key). Five lines above, the code short-circuits if the authenticating user is an API key so that it does not invoke `getRoleDescriptorsList`. If it did, the result would be a list of size 2. This is something es-security talked from time to time for improvement. But it was never formalized. 

The method was actually just `getRoleDescriptor` previously and got changed to `getRoleDescriptorsList` in https://github.com/elastic/elasticsearch/pull/81427/files#diff-65dc248b6cd87fca037bd9e993b2aab3828578878ab92e507a88ce89925b493e with the intention that it will be ready when needed. But since then there has been no progress on this front. 

I don't mind changing it back to just `getRoleDescriptor`. But I'd like to get a second opinion from @n1v0lg. Thanks!

Reply:
:+1: thanks for the extra context, I added a short comment about it in 7fd854d446b.